File: CallByReference/CallByReference6.java
public class CallByReference6 {

    public void swap(int x, int y) {
        int t = x;
        x = y;
        y = t;
        System.out.println("Inside swap(): " + x + " " + y);
    }

    public static void main(String[] args) {
        CallByReference6 obj = new CallByReference6();
        int x = 10, y = 20;

        System.out.println("Before swap(): " + x + " " + y);
        obj.swap(x, y);
        System.out.println("After swap(): " + x + " " + y);
    }
}

File: CallByReference/CallByReference.6.js
function swap(x, y) {
    var t = x;
    x = y;
    y = t;
    console.log('Inside swap():' + x + ' ' + y)
}

let x = 10, y = 20;
console.log('Before swap():' + x + ' ' + y)
swap(x, y)
console.log('After swap():' + x + ' ' + y)
